_includes/comments-providers/discourse.html

```diff
@@ -0,0 +1,13 @@
+{% if site.comments.discourse.server %}
+{% capture canonical %}{{ site.url }}{% if site.permalink contains '.html' %}{{ page.url }}{% else %}{{ page.url | remove:'index.html' | strip_slash }}{% endif %}{% endcapture %}
```

Comment:
Might want to add `{{ site.baseurl }}` after `{{ site.url }}` as the theme supports that and if defined in `_config.yml` will properly assign the wrong page url to `canonical`.